homeassistant/components/tahoma.py

```diff
@@ -78,12 +78,19 @@ def setup(hass, config):
 
 
 def map_tahoma_device(tahoma_device):
-    """Map tahoma classes to Home Assistant types."""
-    if tahoma_device.type.lower().find("shutter") != -1:
+    """Map Tahoma device types to Home Assistant components."""
+    if tahoma_device.type == 'rts:RollerShutterRTSComponent' \
```

Comment:
To clean this up a bit, let's switch to a new pattern. Underneath `TAHOMA_COMPONENTS` at the top of the file, declare a new mapping like this:
```python
TAHOMA_TYPES = {
    'rts:RollerShutterRTSComponent': 'cover',
    'rts:CurtainRTSComponent': 'cover',
    'io:RollerShutterWithLowSpeedManagementIOComponent': 'cover',
    'io:WindowOpenerVeluxIOComponent': 'cover',
    'io:LightIOSystemSensor': 'sensor',
}
```
Then this function can simply be:
```python
return TAHOMA_TYPES.get(tahoma_device.type)
```

We can move the log warning to the `None` check in `setup`.

Reply:
Even better than my suggestion of using a set. Then, I guess ```setup()``` should use the same dict to not log false warnings.